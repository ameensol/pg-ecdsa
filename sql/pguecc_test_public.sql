create extension pgcrypto;
create extension pguecc;

-- sensible input types
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234', 'sha256', 'secp160r1'));
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234'::text, 'sha256', 'secp160r1'));
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234'::bytea, 'sha256', 'secp160r1'));

-- error on invalid hash function
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234', 'bad', 'secp160r1'));

-- ecdsa_is_valid_public_key
select ecdsa_is_valid_public_key(E'\\xDCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB'::bytea, 'secp160r1');
select ecdsa_is_valid_public_key('DCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB', 'secp160r1');
select ecdsa_is_valid_public_key('1234', 'secp160r1');
select ecdsa_is_valid_public_key('00000000000000000000000000000000000000000000000000000000000000000000000000000000', 'secp160r1');

-- ecdsa_is_valid_private_key
select ecdsa_is_valid_private_key(E'\\x0001997CBE1B9E90DAB23322F557E0A2510DA101FA'::bytea, 'secp160r1');
select ecdsa_is_valid_private_key(E'\\x000000000000000000000000000000000000000001'::bytea, 'secp160r1');
select ecdsa_is_valid_private_key('000000000000000000000000000000000000000001', 'secp160r1');
select ecdsa_is_valid_private_key('1234', 'secp160r1');
select ecdsa_is_valid_private_key('000000000000000000000000000000000000000000', 'secp160r1');

-- ecdsa_is_valid_curve
select ecdsa_is_valid_curve('secp256k1');
select ecdsa_is_valid_curve('invalid-curve');

-- ecdsa_make_key
select length((ecdsa_make_key('secp160r1')).public_key);
select ecdsa_is_valid_public_key((ecdsa_make_key('secp160r1')).public_key, 'secp160r1');

select length((ecdsa_make_key('secp160r1')).private_key);
select ecdsa_is_valid_private_key((ecdsa_make_key('secp160r1')).public_key, 'secp160r1');

with key as (select * from ecdsa_make_key('secp160r1'))
select
    ecdsa_verify(
        key.public_key, 'hello, world',
        ecdsa_sign(key.private_key, 'hello, world', 'sha256', 'secp160r1'),
        'sha256', 'secp160r1'
    )
from key;

with key as (select * from ecdsa_make_key('secp256k1'))
select
    ecdsa_verify(
        key.public_key, 'hello, world',
        ecdsa_sign(key.private_key, 'hello, world', 'sha256', 'secp256k1'),
        'sha256', 'secp256k1'
    )
from key;
