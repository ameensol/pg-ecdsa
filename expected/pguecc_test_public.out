create extension pgcrypto;
ERROR:  extension "pgcrypto" already exists
create extension pguecc;
ERROR:  extension "pguecc" already exists
-- sensible input types
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234', 'sha256', 'secp160r1'));
 length 
--------
     80
(1 row)

select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234'::text, 'sha256', 'secp160r1'));
ERROR:  function ecdsa_sign(unknown, text, unknown, unknown) does not exist
LINE 1: select length(ecdsa_sign('0000000000000000000000000000000000...
                      ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234'::bytea, 'sha256', 'secp160r1'));
 length 
--------
     80
(1 row)

-- error on invalid hash function
select length(ecdsa_sign('000000000000000000000000000000000000000000', '1234', 'bad', 'secp160r1'));
ERROR:  Cannot use "bad": No such hash algorithm
CONTEXT:  SQL function "ecdsa_sign" during inlining
-- ecdsa_is_valid_public_key
select ecdsa_is_valid_public_key(E'\\xDCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB'::bytea, 'secp160r1');
 ecdsa_is_valid_public_key 
---------------------------
 t
(1 row)

select ecdsa_is_valid_public_key('DCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB', 'secp160r1');
 ecdsa_is_valid_public_key 
---------------------------
 t
(1 row)

select ecdsa_is_valid_public_key('1234', 'secp160r1');
 ecdsa_is_valid_public_key 
---------------------------
 f
(1 row)

select ecdsa_is_valid_public_key('00000000000000000000000000000000000000000000000000000000000000000000000000000000', 'secp160r1');
 ecdsa_is_valid_public_key 
---------------------------
 f
(1 row)

-- ecdsa_is_valid_private_key
select ecdsa_is_valid_private_key(E'\\x0001997CBE1B9E90DAB23322F557E0A2510DA101FA'::bytea, 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 t
(1 row)

select ecdsa_is_valid_private_key(E'\\x000000000000000000000000000000000000000001'::bytea, 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 t
(1 row)

select ecdsa_is_valid_private_key('000000000000000000000000000000000000000001', 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 t
(1 row)

select ecdsa_is_valid_private_key('1234', 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 f
(1 row)

select ecdsa_is_valid_private_key('000000000000000000000000000000000000000000', 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 f
(1 row)

-- ecdsa_is_valid_curve
select ecdsa_is_valid_curve('secp256k1');
 ecdsa_is_valid_curve 
----------------------
 t
(1 row)

select ecdsa_is_valid_curve('invalid-curve');
 ecdsa_is_valid_curve 
----------------------
 f
(1 row)

-- ecdsa_make_key
select length((ecdsa_make_key('secp160r1')).public_key);
 length 
--------
     80
(1 row)

select ecdsa_is_valid_public_key((ecdsa_make_key('secp160r1')).public_key, 'secp160r1');
 ecdsa_is_valid_public_key 
---------------------------
 t
(1 row)

select length((ecdsa_make_key('secp160r1')).private_key);
 length 
--------
     42
(1 row)

select ecdsa_is_valid_private_key((ecdsa_make_key('secp160r1')).public_key, 'secp160r1');
 ecdsa_is_valid_private_key 
----------------------------
 f
(1 row)

with key as (select * from ecdsa_make_key('secp160r1'))
select
    ecdsa_verify(
        key.public_key, 'hello, world',
        ecdsa_sign(key.private_key, 'hello, world', 'sha256', 'secp160r1'),
        'sha256', 'secp160r1'
    )
from key;
 ecdsa_verify 
--------------
 t
(1 row)

with key as (select * from ecdsa_make_key('secp256k1'))
select
    ecdsa_verify(
        key.public_key, 'hello, world',
        ecdsa_sign(key.private_key, 'hello, world', 'sha256', 'secp256k1'),
        'sha256', 'secp256k1'
    )
from key;
 ecdsa_verify 
--------------
 t
(1 row)

